src/main/java/edu/usf/cutr/grha/utils/GtfsUtils.kt

```diff
@@ -0,0 +1,28 @@
+package edu.usf.cutr.grha.utils
+
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.*
+
+object GtfsUtils {
+
+    @JvmStatic
+     fun getDateFromTimeStamp(timeStamp: String, pattern: String): Date {
+        val format = SimpleDateFormat(pattern, Locale.ENGLISH)
+        format.timeZone = TimeZone.getTimeZone(System.getProperty("user.timezone"))
```

Comment:
This should also be the same timezone for the data specified in the constant (i.e., Chicago), not the timezone on the person's computer that is running the software.

Reply:
If I use the CST timezone, then the time is automatically converted to the EST. For example, if the time on the data is `12:00`, then after the conversion,  the `stop_times` will have the time as `1:00`.

Reply:
I went back to the dataset just to make sure we're doing this right. 

So looking at https://data.cityofchicago.org/Transportation/Transportation-Network-Providers-Trips/m6dm-c72p, it says that the Start and End timestamp fields are "Floating Timestamp", which points to:
https://dev.socrata.com/docs/datatypes/floating_timestamp.html#,

This page says:

>Floating timestamps represent an instant in time with millisecond precision, with no timezone value, encoded as ISO8601 Times with no timezone offset. When writing data, accuracy to only the second is required, but the service will always return precision to the millisecond
...
Datasets will either specify what timezone they should be interpreted in, or you can usually assume they’re in the timezone of the publisher. For example, a dataset published by the City of Chicago will be published in Central Standard Time.

So we're correct in assuming the times are central time to start with, which means if the GTFS dataset agency.txt agency_timezone value is also central time, then we don't need to do anything with the date or time values in terms of projecting to a new timezone or applying daylight savings time. So we really don't need Java to do any operations on these times, as it could introduce artifacts like the one you're mentioning if it's not done correctly (date and time is tricky in Java). 

The correct classes to use in Java when you don't want to apply timezone information is LocalDate and LocalTime (or LocalDateTime, if it's both), as these also reference ISO8601.

In our case, I think we should use LocalDateTime, as we can parse the date and time once and then get the needed values from this class.

Something like the below, although please check my logic:

```Java
DateTimeFormatter formatter = ...; // Define the M/dd/yyyy hh:mm format
LocalDateTime dateTime = LocalDateTime.parse(tncDateTime, formatter);
```

And then for instantiating ServiceDate you can do:

```java
calendarDate.setDate(new ServiceDate(dateTime.getYear(), dateTime.getMonthValue(), dateTime.getDayOfMonth()));
```

And for stop_times.txt you can do:

```java
Duration duration = Duration.between(dateTime.toLocaleDate().atStartOfDay(), dateTime);
stopTime.setArrivalTime(duration.getSeconds());
```

Parsing strings into LocalDate or LocalTime or LocalDateTime can be expensive so please try to do that as little as possible (hence why I think it would be best to parse LocalDateTime once, instead of parsing LocalDate and LocalTime separately).

Please give that a look and see if it would work.

Reply:
Thank you so much for the detailed explanation!